### PR TITLE
Fix delete endpoint

### PR DESCRIPTION
**Pull request by josejulio**
Looks like RHOSE is not answering with the deleted type and instead returning a 404.
Check if we were deleting an endpoint and delete it if that's the case.

### CODE REVIEW

File: backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointReadyChecker.java [hunk 1]
```diff
@@ -92,10 +92,16 @@ public void execute() {
                     em.remove(ep);
                 }
             } catch (WebApplicationException wae) {
-                String path = "GET " + BASE_PATH + "/{bridgeId}/processors/{processorId}";
-                rhoseErrorMetricsRecorder.record(path, wae);
-                Log.warn("Getting data from OB failed", wae);
-                ep.setStatus(EndpointStatus.FAILED);
+                // 404 means that the processor could not be found - but we were deleting an endpoint.
+                // This means is no longer there and we can finish with the delete process
+                if (wae.getResponse().getStatus() == 404 && ep.getStatus().equals(EndpointStatus.DELETING)) {
```
Comment (gwenneg) [inline backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointReadyChecker.java:97]: Did you confirm with the RHOSE team that the 404 is the expected behavior?

Comment (josejulio) [reply backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointReadyChecker.java:97]: Not yet - asked them but waiting for an answer.
This is what I'm currently getting on stage.

Comment (gwenneg) [reply backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointReadyChecker.java:97]: We should probably wait for their answer before merging this PR then.

Comment (josejulio) [reply backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointReadyChecker.java:97]: > if you get 404 it means it does not exist anymore - the resource was deleted successfully

**Comment by josejulio** [comment]
Merging to try on stage

**Pull request merged**